react/react2/week1/week1-homework/app/astronomy/page.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+
+export default async function AstronomyPic() {
+  const API_KEY = "EeSfTBChJ8VYja2AH2LetV3iDAGAaAd7hT14qCfZ";
+
+  const apiResponse = await fetch(
+    `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`,
+    { cache: "no-store" }
+  );
+  if (!apiResponse.ok) {
+    throw new Error("Astronomy picture not found");
+  }
+  const astronomyPic = await apiResponse.json();
+  return (
+    <div className="min-h-screen w-full">
+      {astronomyPic ? (
+        <div className="flex flex-col items-center justify-center gap-5 py-5">
+          <img
+            className="border-2 h-96 w-96"
+            src={astronomyPic.url}
+            alt={astronomyPic.title}
+          ></img>
+          <div className="max-w-2xl bg-[#262b40] py-5 px-6 mx-3">
+            <p className="text-md mx-auto">{astronomyPic.explanation}</p>
+          </div>
+        </div>
+      ) : (
+        <div>
+          <p>Loading the picture....</p>
+        </div>
```

Comment:
_Just a "learning" comment, nothing needs to be changed_

Actually, since it is server rendering, proper "Loading" state will never happen (because we AWAIT for apiResponse and do nothing in the meantime). If there is going to be an empty response, then probably it is not a "Loading" state, but an error

Reply:
Thank you for your inputs. I will remember to have the loading state only for the client components.